enhancements/ignition-spec-dual-support.md

```diff
@@ -0,0 +1,377 @@
+CoreOS (Henceforth RHCOS) package version will also be updated.
+
+This overall migration will take part as a two-phase process:
+Phase 1/OCP 4.4:
+ - MCD gains the ability to process both v2 and v3 configs
+ - MCS gains the ability to translate from v3 configs to v2 configs "on the fly"
+ - MCC attempts to update all v2 configs to v3, leaving those which cannot be converted
+ - Installer and other components generate v3 configs
+
+Phase 2/OCP 4.5:
+ - MCO enforces that all configs are v3 before allowing the CVO to start the update
+ - RHCOS bootimages switches to only accept v3 configs
+
+
+## Motivation
+
+Ignition released v2.0.0 (stable) on Jun 3rd, 2019, which has an updated
+specification format (ignition spec version 3, henceforth “Spec 3”). This
+change includes some important reworks for RHCOS provisioning, most
+importantly the ability to specify and mount other filesystems and fixing
+issues where ignition v0 spec was not declarative. Particularly, this is
+required to support having /var on a separate partition or disk, an important
+requirement for security/compliance purposes in OCP. The existing version on
+RHCOS systems (ignition version v0.33) carries a spec version (spec version 2,
+henceforth “Spec 2”) that is not compatible with Spec 3. Thus we would like to
+update the ignition version on RHCOS/Installer/MCO to make use of the changes.
+
+This proposal will also allow closer alignment with OKD, as OKD will be based
+on Fedora CoreOS (Henceforth FCoS), which is already on, and only supports,
+ignition spec 3. We want to do this in a way that can minimize deltas between
+OKD and OCP.
+
+
+### Goals
+
+#### Phase 1:
+- [ ] A config translator is created to translate from ignition spec 2 to spec 3
+- [ ] A config translator is created to translate from ignition spec 3 to spec 2
+- [ ] The MCO is updated to support both ignition spec 2 and 3, with:
+    - [ ] a translator in the Machine Config Controller that can convert spec 2 cluster configuration to spec 3
+    - [ ] a detector in the Machine Config Server that can serve the correct spec version based on bootimage version
+- [ ] The Openshift installer is updated to generate ignition spec 3 configs
+- [ ] Successfully create new Spec 3 only IPI/UPI clusters
+- [ ] Create a migration method for non-compatible (non-translatable configs) Spec 2 clusters to Spec 3
+- [ ] An alerting mechanism is put in place for outdated and incompatible/non-translatable configs
+- [ ] Docs are updated to reflect the new config version
+
+#### Phase 2:
+- [ ] RHCOS bootimage is updated to accept ignition spec 3 configs
+- [ ] Successfully upgrade existing 4.x clusters to Spec 3 clusters
+
+
+### Non-Goals
+
+ - Support FCoS/OKD directly through this change
+ - API support for MCO, namely switching to RawExt formatted machineconfig
+ objects instead of explicitly referencing ignition, is not considered as
+ part of this proposal
+
+
+## Proposal
+
+This change is multi-component:
```

Comment:
@ericavonb brought up that we might need to sync/communicate with KNI as well (iirc)